ccc/util.cpp

```diff
@@ -1,5 +1,16 @@
 #include "ccc.h"
 
+#include <filesystem>
+#include <fstream>
+#include <iterator>
+#include <vector>
+#include <cstdint>
+
+std::vector<uint8_t> read_file_bin(const std::filesystem::path& path) {
+	std::basic_ifstream<uint8_t> file{ path, std::ios::binary };
+	return { std::istreambuf_iterator<uint8_t>{file}, {} };
```

Comment:
For some reason this is crashing for me:

    terminate called after throwing an instance of 'std::bad_cast'
      what():  std::bad_cast
    
    Program received signal SIGABRT, Aborted.
    __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
    50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
    (gdb) bt
    #0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
    #1  0x00007ffff7bb6859 in __GI_abort () at abort.c:79
    #2  0x00007ffff7e3c951 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    #3  0x00007ffff7e4847c in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    #4  0x00007ffff7e484e7 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    #5  0x00007ffff7e487ed in __cxa_rethrow () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    #6  0x000055555557f78a in void std::vector<unsigned char, std::allocator<unsigned char> >::_M_range_initialize<std::istreambuf_iterator<unsigned char,     std::char_traits<unsigned char> > >(std::istreambuf_iterator<unsigned char, std::char_traits<unsigned char> >, std::istreambuf_iterator<unsigned char,     std::char_traits<unsigned char> >, std::input_iterator_tag) ()
    #7  0x000055555557f43f in std::vector<unsigned char, std::allocator<unsigned char> >::vector<std::istreambuf_iterator<unsigned char, std::char_traits<unsigned char>     >, void>(std::istreambuf_iterator<unsigned char, std::char_traits<unsigned char> >, std::istreambuf_iterator<unsigned char, std::char_traits<unsigned char> >,     std::allocator<unsigned char> const&) ()
    #8  0x000055555557eece in read_file_bin(std::filesystem::__cxx11::path const&) ()
    #9  0x000055555557687c in read_program_image(std::filesystem::__cxx11::path) ()
    #10 0x0000555555571cbc in main ()

I must admit I don't know much about the C++ stream iterators so I don't know why, maybe just try a different implementation.